k8s/dashboard.go

```diff
@@ -0,0 +1,99 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"time"
+	"tyk/tyk/bootstrap/pkg/constants"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// RestartDashboard restarts Tyk Dashboard Deployment. Restarting Tyk Dashboard is needed to apply
+// CName.
+func (c *Client) RestartDashboard() error {
+	if c.appArgs.K8s.DashboardDeploymentName == "" {
+		ls := metav1.LabelSelector{MatchLabels: map[string]string{
+			constants.TykBootstrapLabel: constants.TykBootstrapDashboardDeployLabel,
+		}}
+
+		deployments, err := c.clientSet.
+			AppsV1().
+			Deployments(c.appArgs.K8s.ReleaseNamespace).
+			List(
+				context.TODO(),
+				metav1.ListOptions{
+					LabelSelector: labels.Set(ls.MatchLabels).String(),
+				},
+			)
+		if err != nil {
+			return fmt.Errorf("failed to list Tyk Dashboard Deployment, err: %v", err)
+		}
+
+		for i := range deployments.Items {
+			c.appArgs.K8s.DashboardDeploymentName = deployments.Items[i].ObjectMeta.Name
```

Comment:
Since we are iterating over the items, DashboardDeploymentName will be overridden and ultimately last deployment will be used.
Why don't we skip loop and directly access last element?

Reply:
updated as you suggestedf